zulipterminal/ui_tools/buttons.py

```diff
@@ -288,11 +288,36 @@ def handle_link(self, *_: Any) -> None:
         """
         if self.is_internal_link:
             if '/#narrow/' in self.link:
-                # TODO: Handle internal link.
-                pass
+                self.handle_narrow()
             elif '/user_uploads/' in self.link:
                 # TODO: Open media in the native viewer.
                 pass
         else:
             # TODO: Open external link in the web browser.
             pass
+
+    def handle_narrow(self) -> None:
+        """
+        Handles stream narrow.
+        """
+        # TODO: Handle topic narrow.
+        link_split = self.link.split('/')
+
+        self.stream_id = int(link_split[5].split('-')[0])
+
+        # Return if the concerned stream is unsubscribed.
+        if self.stream_id not in self.model.stream_dict:
+            text = ' You are not subscribed to the stream'
+            self.view.set_footer_text(text, duration=3)
+            return
+
+        self.stream_name = self.model.stream_dict[self.stream_id]['name']
+
+        if '/stream/' in self.link:
+            assert hasattr(self, 'stream_id')
+            assert hasattr(self, 'stream_name')
+            self.controller.narrow_to_stream(self)
+
+        # Exit popup if the narrow was successful.
+        if hasattr(self.controller.loop.widget, '_Overlay__super'):
```

Comment:
This seems rather cryptic - is there a better expression for this? What narrow failure are we expecting? Should the narrow methods return a value instead?

Reply:
Fixed with a simple expression. I was only expecting failure from the code that I previously had here (early return statements), but I have improved the structure now.

Re narrow methods, I'd be inclined towards handle failures, if any, in the narrow methods itself than returning a value as a separate amendment.